daemon.go

```diff
@@ -157,23 +158,24 @@ func clearUnixSockets(path ma.Multiaddr) error {
 	if err := os.Remove(c.Value()); err != nil {
 		return err
 	}
-	
+
 	return nil
 }
 
 func (d *Daemon) Close() error {
+	var merr multierror.Error
```

Comment:
make this a pointer so that you don't have to copy with dereferencing.

Reply:
Done

Reply:
see https://github.com/libp2p/go-libp2p-swarm/pull/115/files for an example of usage.